### PR TITLE
Add a welcome message to login

### LINKED ISSUE #3152: Make it more obvious *why* you would log in
At the moment, we just ask folks to login without much context about why. And we don't do a great job of listing [alternative self-managed options](https://www.pulumi.com/docs/intro/concepts/state/#self-managed-backend). We should fix this.

### PR DESCRIPTION
**Pull request by lukehoban**
Fixes #3152.

Here's an initial proposal for updates to the `pulumi login` experience to provide more details on why you need to login, what login options you have, and details on what information is *not* shared with the Pulumi Service.

Before (pre-login):
Screen Shot 2021-02-13 at 5 40 16 AM

Before (post-login):
Screen Shot 2021-02-13 at 5 39 58 AM

After (pre-login):
Screen Shot 2021-02-13 at 5 41 13 AM

After (post-login):
Screen Shot 2021-02-13 at 5 41 30 AM

Full text:

```

Welcome to Pulumi!

  Pulumi helps you create, deploy, and manage infrastructure on any cloud using
  your favorite language.  To manage your cloud infrastructure, Pulumi stores the
  state of your cloud deployments in a backend of your choosing.

  To use the default Pulumi Service backend, login below.  If you would prefer to
  manage state yourself, you can choose from a variety of self-managed Pulumi
  backends. Run `pulumi login --help` for alternative login options.

  The Pulumi Service backend provides transparent, robust and simple storage for your
  Pulumi state. All secret values are encrypted in the state, and you can
  bring your own encryption provider for secrets. Pulumi deployments are always executed
  locally where the `pulumi` CLI is being invoked, so no ambient cloud credentials,
  source code, or other sensitive data will ever be shared with the Pulumi Service
  backend. For more details - see https://www.pulumi.com/docs/intro/concepts/state/.

Hit <ENTER> to log in using your browser, or else visit https://app.pulumi.com/account/tokens
to manually provision an access token and enter it here:
```

```
You are now logged in!

  You can get started with Pulumi at:

      https://www.pulumi.com/docs/get-started/

  Tip of the day: Resources you create with Pulumi are given unique names (a randomly
  generated suffix) by default. To learn more about auto-naming or customizing resource
  names see https://www.pulumi.com/docs/intro/concepts/programming-model/#autonaming.


Logged in to pulumi.com as lukehoban (https://app.pulumi.com/lukehoban)
```

### CODE REVIEW

File: pkg/backend/httpstate/backend.go [hunk 1]
```diff
@@ -274,17 +274,38 @@ func Login(ctx context.Context, d diag.Sink, cloudURL string, opts display.Optio
...
+			colors.Highlight(
+				"  To use the default Pulumi Service backend, login below.  If you would prefer to",
+				"login below",
+				colors.BrightCyan+colors.Bold,
+			),
+			"  manage state yourself, you can choose from a variety of self-managed Pulumi",
+			"  backends. Run `pulumi login --help` for alternative login options.",
+			"",
+			colors.Highlight(
+				"  The Pulumi Service backend provides transparent, robust and simple storage for your",
```
Comment (infin8x) [inline pkg/backend/httpstate/backend.go:295]:
```suggestion
				"  The Pulumi Service backend provides transparent, robust, and simple storage for your",
```

File: pkg/backend/httpstate/backend.go [hunk 2]
```diff
@@ -274,17 +274,38 @@ func Login(ctx context.Context, d diag.Sink, cloudURL string, opts display.Optio
...
+				"Pulumi Service",
+				colors.Underline+colors.Bold,
+			),
+			"  Pulumi state. All secret values are encrypted in the state, and you can",
+			"  bring your own encryption provider for secrets. Pulumi deployments are always executed",
```
Comment (infin8x) [inline pkg/backend/httpstate/backend.go:300]:
```suggestion
			"  Pulumi state. All secret values in the state are encrypted in the Pulumi Service; you can also",
			"  bring your own encryption provider for secrets. Pulumi deployments are always executed",
```

File: pkg/backend/httpstate/backend.go [hunk 3]
```diff
@@ -274,17 +274,38 @@ func Login(ctx context.Context, d diag.Sink, cloudURL string, opts display.Optio
...
+			"  locally where the `pulumi` CLI is being invoked, so no ambient cloud credentials,",
```
Comment (infin8x) [inline pkg/backend/httpstate/backend.go:301]: I can figure out what "ambient cloud credentials" means based on our past discussions, but will our customers know what we mean?

File: pkg/backend/httpstate/backend.go [hunk 4]
```diff
@@ -274,17 +274,38 @@ func Login(ctx context.Context, d diag.Sink, cloudURL string, opts display.Optio
...
+			"  source code, or other sensitive data will ever be shared with the Pulumi Service",
+			"  backend. For more details - see https://www.pulumi.com/docs/intro/concepts/state/.",
```
Comment (infin8x) [inline pkg/backend/httpstate/backend.go:303]: Seems like this might be worth highlighting, maybe bold, or green text, or similar?

**Review by infin8x — APPROVED** [review]
Looks great! Added some suggestions for readability, but otherwise 🚢  it.

File: pkg/backend/httpstate/backend.go (hunk 3 above)
Comment (lukehoban) [reply pkg/backend/httpstate/backend.go:301]: I wasn't sure what to say here.  Could replace with just "cloud credentials" perhaps?  or "cloud credentials from your local machine environment"?

File: pkg/backend/httpstate/backend.go (hunk 4 above)
Comment (infin8x) [inline pkg/backend/httpstate/backend.go:303]:
```suggestion
			"  locally where the `pulumi` CLI is being invoked. No source code and no",
			"  no credentials, except your Pulumi login, are sent to the Pulumi Service",
			"  backend. For more details - see https://www.pulumi.com/docs/intro/concepts/state/.",
```

File: pkg/backend/httpstate/backend.go (hunk 3 above)
Comment (infin8x) [reply pkg/backend/httpstate/backend.go:301]: Tried a suggestion a bit further down (in a different comment since I had to span multiple lines.

**Review by infin8x — APPROVED** [review]

**Comment by stack72** [comment]
@lukehoban / @infin8x what's the status of this - can we rebase and merge it?

**Comment by infin8x** [comment]
@stack72 I'm happy with these changes. I took the liberty of doing the rebase and the `CHANGELOG` (apologies if this is bad form on someone else's PR - LMK if so). If @lukehoban is 👍🏻, let's merge it.

**Comment by lukehoban** [comment]
Sorry for delay - after some offline discussion we will be making a different change to address this.  I will follow up with an updated PR here.  I'll close this one out for now.

**Pull request closed without merging**